### PR TITLE
Adjust style to be similar with origin

### LINKED ISSUE #14885: FormItem使用extra时,提示文案与表单项组件之间的间距太大
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/form-cn/#components-form-demo-register](https://ant.design/components/form-cn/#components-form-demo-register)

[https://ant.design/components/form-cn/#components-form-demo-validate-other](https://ant.design/components/form-cn/#components-form-demo-validate-other)

### Steps to reproduce

image

image

### What is expected?
使用extra和help时, 样式表现正常

### What is actually happening?
使用extra时, 存在额外的间距.

| Environment | Info |
|---|---|
| antd | 3.13.3 |
| React | 16.6.0 |
| System | macOS 10.14.2 |
| Browser | Chrom 72 |

---

### PR DESCRIPTION
**Pull request by zombieJ**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

Fix style not sync:
3.13.2: https://codesandbox.io/s/381x3nvmqq
3.13.3: https://codesandbox.io/s/73rmyyl6rq

fix: #14885
ref: #14778

### Changelog description (Optional if not new feature)

> 1. Fix FormItem margin not correct.
> 2. 修复 FormItem margin 不正确的问题。

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by ztplz — APPROVED** [review]

**Comment by zombieJ** [comment]
还有一个 issue #14885，顺道一起修了吧

File: components/form/FormItem.tsx [hunk 1]
```diff
@@ -157,25 +157,36 @@ export default class FormItem extends React.Component<FormItemProps, any> {
...
+  }
+
+  renderHelpHolder(prefixCls: string) {
+    const { style } = this.props;
+    if (this.helpShow || (style && ('margin' in style || 'marginBottom' in style))) {
+      return null;
+    }
+
+    return (
+      <div className={`${prefixCls}-explain-holder`} aria-hidden="true">
```
Comment (afc163) [inline components/form/FormItem.tsx]: 是不是也不需要 `.ant-form-explain-holder`，直接保持 `.ant-form-explain` 的 dom 结构，然后用 `ant-form-explain:empty:before` 做占位。

Comment (zombieJ) [reply components/form/FormItem.tsx]: 因为有 extra 的关系，占位需要放在最后面才行：



另外，extra 也是有了才渲染。直接在最后面设置一个占位 span 相比判断有无 extra 设置伪类逻辑会更好懂些。

**Review by afc163 — APPROVED** [review]

**Comment by VoliBearCat** [comment]
image
image

建议 .ant-form-explain 和 .ant-form-extra 一样拥有padding-top: 4px;
@zombieJ @afc163

**Comment by bgold0** [comment]
How do we disable this explain holder? It completely broke my entire site's styling.

**Comment by zombieJ** [comment]
> How do we disable this `.ant-form-explain-holder`? It completely broke my entire site's styling. Why is this being accepted?

Hi @bgold0 , could you help to provide an online reproduce?

**Comment by LeoHuiyi** [comment]
项目使用@form-item-margin-bottom 这个变量定制主题，布局还是有问题

**Pull request merged**